From 15af882cff915eada828114bdbcf2b7d19595b67 Mon Sep 17 00:00:00 2001
From: Gergely Nagy <algernon@madhouse-project.org>
Date: Thu, 4 Dec 2014 10:29:36 +0100
Subject: [PATCH] hostname: Demote the FQDN warning to a verbose message

On a lot of desktop machines (and in private networks), the hostname
will not be fully qualified, so the warning is just noise. Demote it to
a message only displayed in verbose mode.

This fixes the other half of #329.

Reported-by: Peter Czanik <czanik@balabit.hu>
Signed-off-by: Gergely Nagy <algernon@madhouse-project.org>
---
 lib/hostname.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/hostname.c b/lib/hostname.c
index b65d8a1..5af34ec 100644
--- a/lib/hostname.c
+++ b/lib/hostname.c
@@ -135,8 +135,8 @@ detect_local_fqdn_hostname(void)
       hostname = get_local_fqdn_hostname_from_dns();
       if (!hostname)
         {
-          msg_warning("Unable to detect fully qualified hostname for localhost, use_fqdn() will use the short hostname",
-                     NULL);
+          msg_verbose("Unable to detect fully qualified hostname for localhost, use_fqdn() will use the short hostname",
+                      NULL);
           hostname = get_local_hostname_from_system();
         }
     }
From 1ddfe7878f1089b20e854f30a6bdb9acb5c6f423 Mon Sep 17 00:00:00 2001
From: Gergely Nagy <algernon@madhouse-project.org>
Date: Thu, 4 Dec 2014 10:27:28 +0100
Subject: [PATCH] system-source: Demote the json-parser() warning to a debug
 message

Not having the json module installed by default is fairly common, @cim
messages on the other hand, are not. To make startup less annoying for a
large number of users, demote the warning about json-parser() missing to
a debug message.

This addresses one part of #329.

Reported-by: Peter Czanik <czanik@balabit.hu>
Signed-off-by: Gergely Nagy <algernon@madhouse-project.org>
---
 modules/system-source/system-source.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules/system-source/system-source.c b/modules/system-source/system-source.c
index b9580e4..7228dd4 100644
--- a/modules/system-source/system-source.c
+++ b/modules/system-source/system-source.c
@@ -323,8 +323,8 @@ system_generate_cim_parser(GlobalConfig *cfg, GString *sysblock)
 
   if (!_is_json_parser_available(cfg))
     {
-      msg_warning_once("system(): json-parser() is missing, skipping the automatic JSON parsing of messages submitted via syslog(3), Please install the json module",
-                       NULL);
+      msg_debug("system(): json-parser() is missing, skipping the automatic JSON parsing of messages submitted via syslog(3), Please install the json module",
+                NULL);
       return;
     }
 
